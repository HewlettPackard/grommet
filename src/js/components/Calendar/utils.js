
// Utility functions for the Calendar.
// Just what's needed to avoid having to include a dependency like momentjs.

const DAY_MILLISECONDS = 24 * 60 * 60 * 1000;

export const addDays = (date, days) =>
  (new Date(date.getTime() + (DAY_MILLISECONDS * days)));

export const subtractDays = (date, days) => addDays(date, -days);

export const addMonths = (date, months) => {
  const result = new Date(date);
  const years = Math.floor((date.getMonth() + months) / 12);
  result.setFullYear(date.getFullYear() + years);
  result.setMonth((date.getMonth() + months) % 12);
  return result;
};

export const subtractMonths = (date, months) => addMonths(date, -months);

export const sameDay = (date1, date2) => (
  date1.getFullYear() === date2.getFullYear() &&
  date1.getMonth() === date2.getMonth() &&
  date1.getDate() === date2.getDate()
);

export const sameDayOrAfter = (date1, date2) =>
  (date1.getFullYear() > date2.getFullYear() ||
    (date1.getFullYear() === date2.getFullYear() && (
      date1.getMonth() > date2.getMonth() || (
        date1.getMonth() === date2.getMonth() && (date1.getDate() >= date2.getDate())
      )
    ))
  );

export const sameDayOrBefore = (date1, date2) =>
  (date1.getFullYear() < date2.getFullYear() ||
    (date1.getFullYear() === date2.getFullYear() && (
      date1.getMonth() < date2.getMonth() || (
        date1.getMonth() === date2.getMonth() && (date1.getDate() <= date2.getDate())
      )
    ))
  );

export const daysApart = (date1, date2) =>
  Math.floor((date1.getTime() - date2.getTime()) / DAY_MILLISECONDS);

// betweenDates takes and array of two elements and checks if the
// supplied date lies between them, inclusive.
// returns 2 if exact match to one end, 1 if between, undefined otherwise
export const betweenDates = (date, dates) => {
  let result;
  if (dates) {
    const [from, to] = dates.map(d => new Date(d));
    if (sameDay(date, from) || sameDay(date, to)) {
      result = 2;
    } else if (sameDayOrAfter(date, from) && sameDayOrBefore(date, to)) {
      result = 1;
    }
  } else {
    result = 1;
  }
  return result;
};

// withinDates takes and array of string dates or 2 element arrays and
// checks whether the supplied date matches any string or is between
// any dates in arrays.
// returns 2 if exact match, 1 if between, undefined otherwise
export const withinDates = (date, dates) => {
  let result;
  if (dates) {
    if (Array.isArray(dates)) {
      dates.some((d) => {
        if (typeof d === 'string') {
          if (sameDay(date, new Date(d))) {
            result = 2;
          }
        } else {
          result = betweenDates(date, d);
        }
        return result;
      });
    } else if (sameDay(date, new Date(dates))) {
      result = 2;
    }
  }
  return result;
};
