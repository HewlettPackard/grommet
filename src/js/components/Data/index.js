export { Data } from './Data';
export { DataForm } from './DataForm';
