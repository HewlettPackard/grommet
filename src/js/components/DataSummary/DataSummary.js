import React, { useContext } from 'react';
import { Text } from '../Text';
import { DataContext } from '../../contexts/DataContext';
import { MessageContext } from '../../contexts/MessageContext';

import { DataSummaryPropTypes } from './propTypes';

export const DataSummary = ({ messages, ...rest }) => {
  const { format } = useContext(MessageContext);
  const {
    filteredTotal,
    messages: dataMessages,
    total,
  } = useContext(DataContext);

  let messageId;
  if (total !== filteredTotal) {
    if (filteredTotal === 1) messageId = 'dataSummary.filteredSingle';
    else messageId = 'dataSummary.filtered';
  } else if (total === 1) messageId = 'dataSummary.totalSingle';
  else messageId = 'dataSummary.total';

  // helps account for cases like 0 results of 1 item
  const items = format({
    id: total === 1 ? 'dataSummary.itemsSingle' : 'dataSummary.items',
    messages: messages || dataMessages?.dataSummary,
  });

  return (
    <Text margin={{ vertical: 'xsmall' }} {...rest}>
      {format({
        id: messageId,
        messages: messages || dataMessages?.dataSummary,
        values: {
          filteredTotal,
          total,
          items,
        },
      })}
    </Text>
  );
};

DataSummary.propTypes = DataSummaryPropTypes;
