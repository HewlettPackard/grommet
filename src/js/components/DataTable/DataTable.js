import React, { Component } from 'react';

import { Header } from './Header';
import { Footer } from './Footer';
import { Body } from './Body';
import { GroupedBody } from './GroupedBody';
import { buildState } from './buildState';
import { StyledDataTable } from './StyledDataTable';

const contexts = ['header', 'body', 'footer'];

const normalizeProp = (prop, context) => {
  if (prop) {
    if (prop[context]) return prop[context];
    if (contexts.some(c => prop[c])) return undefined;
    return prop;
  }
  return undefined;
};

class DataTable extends Component {
  static defaultProps = {
    columns: [],
    data: [],
    step: 50,
  };

  state = {};

  static getDerivedStateFromProps(nextProps, prevState) {
    return buildState(nextProps, prevState);
  }

  onFiltering = property => {
    this.setState({ filtering: property });
  };

  onFilter = (property, value) => {
    /* eslint-disable-next-line react/prop-types */
    const { onSearch } = this.props;
    const { filters } = this.state;
    const nextFilters = { ...filters };
    nextFilters[property] = value;
    this.setState({ filters: nextFilters });

    // Let caller know about search, if interested
    if (onSearch) {
      onSearch(nextFilters);
    }
  };

  onSort = property => () => {
    const { sort } = this.state;
    const ascending =
      sort && property === sort.property ? !sort.ascending : true;
    this.setState({ sort: { property, ascending } });
  };

  onToggleGroup = groupValue => () => {
    const { groupState } = this.state;
    const { groupBy } = this.props;
    const nextGroupState = { ...groupState };
    nextGroupState[groupValue] = {
      ...nextGroupState[groupValue],
      expanded: !nextGroupState[groupValue].expanded,
    };
    this.setState({ groupState: nextGroupState });
    if (groupBy.onExpand) {
      const expandedKeys = Object.keys(nextGroupState).filter(
        k => nextGroupState[k].expanded,
      );
      groupBy.onExpand(expandedKeys);
    }
  };

  onToggleGroups = () => {
    const { groupState } = this.state;
    const { groupBy } = this.props;
    const expanded =
      Object.keys(groupState).filter(k => !groupState[k].expanded).length === 0;
    const nextGroupState = {};
    Object.keys(groupState).forEach(k => {
      nextGroupState[k] = { ...groupState[k], expanded: !expanded };
    });
    this.setState({ groupState: nextGroupState });
    if (groupBy.onExpand) {
      const expandedKeys = Object.keys(nextGroupState).filter(
        k => nextGroupState[k].expanded,
      );
      groupBy.onExpand(expandedKeys);
    }
  };

  onResize = property => width => {
    const { widths } = this.state;
    const nextWidths = { ...(widths || {}) };
    nextWidths[property] = width;
    this.setState({ widths: nextWidths });
  };

  render() {
    const {
      background,
      border,
      /* eslint-disable-next-line react/prop-types */
      columns,
      data: propsData,
      groupBy,
      onMore,
      replace,
      pad,
      resizeable,
      rowProps,
      size,
      sortable,
      step,
      onClickRow, // removing unknown DOM attributes
      onSearch, // removing unknown DOM attributes
      ...rest
    } = this.props;
    const {
      data,
      filtering,
      filters,
      footerValues,
      groups,
      groupState,
      primaryProperty,
      showFooter,
      sort,
      widths,
    } = this.state;

    if (size && resizeable) {
      console.warn('DataTable cannot combine "size" and "resizeble".');
    }

    return (
      <StyledDataTable {...rest}>
        <Header
          background={normalizeProp(background, 'header')}
          border={normalizeProp(border, 'header')}
          columns={columns}
          filtering={filtering}
          filters={filters}
          groups={groups}
          groupState={groupState}
          pad={normalizeProp(pad, 'header')}
          size={size}
          sort={sort}
          widths={widths}
          onFiltering={this.onFiltering}
          onFilter={this.onFilter}
          onResize={resizeable ? this.onResize : undefined}
          onSort={sortable ? this.onSort : undefined}
          onToggle={this.onToggleGroups}
        />
        {groups ? (
          <GroupedBody
            background={normalizeProp(background, 'body')}
            border={normalizeProp(border, 'body')}
            columns={columns}
            groupBy={groupBy.property ? groupBy.property : groupBy}
            groups={groups}
            groupState={groupState}
            pad={normalizeProp(pad, 'body')}
            primaryProperty={primaryProperty}
            onToggle={this.onToggleGroup}
          />
        ) : (
          <Body
            background={normalizeProp(background, 'body')}
            border={normalizeProp(border, 'body')}
            columns={columns}
            data={data}
            onMore={onMore}
            replace={replace}
            onClickRow={onClickRow}
            pad={normalizeProp(pad, 'body')}
            primaryProperty={primaryProperty}
            rowProps={rowProps}
            size={size}
            step={step}
          />
        )}
        {showFooter && (
          <Footer
            background={normalizeProp(background, 'footer')}
            border={normalizeProp(border, 'footer')}
            columns={columns}
            footerValues={footerValues}
            groups={groups}
            pad={normalizeProp(pad, 'footer')}
            primaryProperty={primaryProperty}
            size={size}
          />
        )}
      </StyledDataTable>
    );
  }
}

let DataTableDoc;
if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  DataTableDoc = require('./doc').doc(DataTable);
}
const DataTableWrapper = DataTableDoc || DataTable;

export { DataTableWrapper as DataTable };
