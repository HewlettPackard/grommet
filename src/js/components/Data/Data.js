import React, { useEffect, useMemo, useState } from 'react';
import { Box } from '../Box';
import { DataFilters } from '../DataFilters';
import { DataSearch } from '../DataSearch';
import { DataSummary } from '../DataSummary';
import { Toolbar } from '../Toolbar';
import { DataContext } from '../../contexts/DataContext';
import { DataPropTypes } from './propTypes';
import { filter } from './filter';

const defaultView = {
  search: '',
  properties: {},
};

export const Data = ({
  children,
  data: dataProp,
  id = 'data',
  messages,
  onView,
  properties,
  toolbar,
  total,
  updateOn = 'submit',
  view: viewProp,
  ...rest
}) => {
  const [view, setView] = useState(viewProp);
  useEffect(() => setView(viewProp), [viewProp]);

  const data = useMemo(() => {
    if (onView) return dataProp;
    return filter(dataProp, view, properties);
  }, [dataProp, onView, properties, view]);

  // what we use for DataContext value
  const contextValue = useMemo(() => {
    const result = { id, messages, properties, updateOn, view };

    if (
      view?.search ||
      view?.sort ||
      (view?.properties && Object.keys(view.properties).length)
    ) {
      result.clearFilters = () => {
        const nextView = defaultView;
        setView(nextView);
        if (onView) onView(nextView);
      };
    }

    result.onView = (nextView) => {
      setView(nextView);
      if (onView) onView(nextView);
    };

    result.data = data;
    result.unfilteredData = dataProp;
    result.total = total !== undefined ? total : dataProp.length;

    return result;
  }, [data, dataProp, id, messages, onView, properties, total, updateOn, view]);

  let toolbarContent;
  if (toolbar) {
    toolbarContent = [
      <Toolbar key="toolbar">
        {(toolbar === true || toolbar === 'search') && <DataSearch />}
        {(toolbar === true || toolbar === 'filters') && <DataFilters drop />}
      </Toolbar>,
      <DataSummary key="summary" />,
    ];
  }

  return (
    <DataContext.Provider value={contextValue}>
      <Box id={id} flex={false} {...rest}>
        {toolbarContent}
        {children}
      </Box>
    </DataContext.Provider>
  );
};

Data.propTypes = DataPropTypes;
