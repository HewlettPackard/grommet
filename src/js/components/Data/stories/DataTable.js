import React from 'react';

import {
  DataFilters,
  DataFilter,
  DataSearch,
  DataSort,
  DataSummary,
  DataTable,
  Grid,
  Notification,
  Toolbar,
} from 'grommet';

import { Data } from '../Data';
import { columns, DATA } from '../../DataTable/stories/data';

export const Table = () => (
  // Uncomment <Grommet> lines when using outside of storybook
  // <Grommet theme={...}>
  <Grid
    flex={false}
    pad="large"
    columns={[['small', 'large']]}
    justifyContent="center"
    gap="large"
  >
    <Notification
      status="info"
      message="Data is in 'beta'. The API surface is subject to change."
    />
    <Data data={DATA}>
      <Toolbar>
        <DataSearch />
        <DataFilters drop>
          <DataFilter property="location" />
          <DataSort />
        </DataFilters>
      </Toolbar>
      <DataSummary />
      <DataTable columns={columns} />
    </Data>
  </Grid>
  // </Grommet>
);

Table.storyName = 'DataTable';

Table.args = {
  full: true,
};

export default {
  title: 'Layout/Data/DataTable',
};
