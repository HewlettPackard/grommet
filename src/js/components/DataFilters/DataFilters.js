import React, { Children, useContext, useState } from 'react';
import { Filter } from 'grommet-icons/icons/Filter';
import { Box } from '../Box';
import { Button } from '../Button';
import { DataFilter } from '../DataFilter';
import { DataForm } from '../Data';
import { DropButton } from '../DropButton';
import { Header } from '../Header';
import { Heading } from '../Heading';
import { DataContext } from '../../contexts/DataContext';
import { MessageContext } from '../../contexts/MessageContext';
import { DataFiltersPropTypes } from './propTypes';

const dropProps = {
  align: { top: 'bottom', right: 'right' },
};

export const DataFilters = ({ drop, children, heading, ...rest }) => {
  const { clearFilters, data, messages, properties, view } =
    useContext(DataContext);
  const { format } = useContext(MessageContext);
  const [showContent, setShowContent] = useState();
  const controlled = drop;

  const clearControl = clearFilters && (
    <Box flex={false}>
      <Button
        label={format({
          id: 'dataFilters.clear',
          messages: messages?.dataFilters,
        })}
        onClick={clearFilters}
      />
    </Box>
  );

  let filters;
  if (Children.count(children) === 0) {
    let filtersFor;
    if (!properties && data && data.length) filtersFor = Object.keys(data[0]);
    else if (Array.isArray(properties)) filtersFor = properties;
    else if (typeof properties === 'object')
      filtersFor = Object.keys(properties);
    else filtersFor = [];
    filters = filtersFor.map((property) => (
      <DataFilter key={property} property={property} />
    ));
  }

  const content = (
    <DataForm
      pad={controlled ? 'medium' : undefined}
      gap="small"
      onDone={() => setShowContent(false)}
      {...(!controlled ? rest : {})}
    >
      {!drop && (
        <Header>
          <Heading margin="none" level={2} size="small">
            {heading ||
              format({
                id: 'dataFilters.heading',
                messages: messages?.dataFilters,
              })}
          </Heading>
          {!controlled && clearControl}
        </Header>
      )}
      {filters}
      {children}
    </DataForm>
  );

  if (!controlled) return content;

  let badge = 0;
  if (view?.properties) badge += Object.keys(view.properties).length;
  if (view?.search) badge += 1;
  if (view?.sort) badge += 1;
  if (!badge) badge = undefined;

  // drop
  const control = (
    <DropButton
      aria-label={format({
        id: 'dataFilters.open',
        messages: messages?.dataFilters,
      })}
      kind="toolbar"
      icon={<Filter />}
      dropProps={dropProps}
      dropContent={content}
      badge={badge}
      open={showContent}
      onOpen={() => setShowContent(undefined)}
      onClose={() => setShowContent(undefined)}
    />
  );

  return (
    <Box flex={false} direction="row" gap="small" {...rest}>
      {control}
      {clearControl}
    </Box>
  );
};

DataFilters.propTypes = DataFiltersPropTypes;
