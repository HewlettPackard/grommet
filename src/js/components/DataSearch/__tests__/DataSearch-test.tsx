import React from 'react';
import { act, fireEvent, render } from '@testing-library/react';
import 'jest-styled-components';
import { Data } from '../../Data';
import { DataFilters } from '../../DataFilters';
import { Grommet } from '../../Grommet';
import { DataSummary } from '../../DataSummary';
import { DataTable } from '../../DataTable';

import { DataSearch } from '..';
import { createPortal, expectPortal } from '../../../utils/portal';

const data = [{ name: 'a' }, { name: 'b' }];

describe('DataSearch', () => {
  window.scrollTo = jest.fn();
  beforeEach(createPortal);

  test('renders', () => {
    const { container } = render(
      <Grommet>
        <Data data={data}>
          <DataFilters>
            <DataSearch />
          </DataFilters>
        </Data>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('no Data', () => {
    const { container } = render(
      <Grommet>
        <DataSearch />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('drop', () => {
    jest.useFakeTimers();

    const { container, getByRole } = render(
      <Grommet>
        <Data id="test-data" data={data}>
          <DataSearch drop />
        </Data>
      </Grommet>,
    );

    expect(getByRole('button', { name: 'Open search' })).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();

    fireEvent.click(getByRole('button', { name: 'Open search' }));
    // advance timers so drop can open
    act(() => jest.advanceTimersByTime(200));

    // snapshot on drop
    expectPortal('test-data--search-control').toMatchSnapshot();
  });
});

test('enter', async () => {
  jest.useFakeTimers();
  const { asFragment, getByTestId, getByText } = render(
    <Grommet>
      <Data data={[
            { a: 'one', b: 1 },
            { a: 'two', b: 2 },
          ]} updateOn="change">
        <DataSearch data-testid="input_submit" updateOn="submit" />
        <DataSummary />
        <DataTable
          columns={[
            { property: 'a', header: 'A' },
            { property: 'b', header: 'B' },
          ]}
          
        />
      </Data>
    </Grommet>,
  );
  const submitButton = getByTestId('input_submit');
  expect(submitButton).toBeTruthy();
  fireEvent.change(submitButton, { target: { value: 'one' } });
  const event = new KeyboardEvent('keydown', { keyCode: 32 });
  act(() => jest.advanceTimersByTime(200));
  submitButton.dispatchEvent(event);
  // make sure one is in the table
  expect(getByText('one')).toBeTruthy();
  // make sure two is not in the table
  expect(getByText('two')).toBeNull();
  // make sure the snapshot is correct
  expect(asFragment()).toMatchSnapshot();
}, 20000);
