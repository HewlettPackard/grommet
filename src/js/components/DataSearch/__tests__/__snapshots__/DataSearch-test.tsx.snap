// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSearch drop 1`] = `
.c3 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c3 g {
  fill: inherit;
  stroke: inherit;
}

.c3 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c3 *[stroke*="#"],
.c3 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c3 *[fill-rule],
.c3 *[FILL-RULE],
.c3 *[fill*="#"],
.c3 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c2 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
  line-height: 0;
  padding: 12px;
}

.c2:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c2:focus > circle,
.c2:focus > ellipse,
.c2:focus > line,
.c2:focus > path,
.c2:focus > polygon,
.c2:focus > polyline,
.c2:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c2:focus::-moz-focus-inner {
  border: 0;
}

.c2:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c2:focus:not(:focus-visible) > circle,
.c2:focus:not(:focus-visible) > ellipse,
.c2:focus:not(:focus-visible) > line,
.c2:focus:not(:focus-visible) > path,
.c2:focus:not(:focus-visible) > polygon,
.c2:focus:not(:focus-visible) > polyline,
.c2:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c2:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <div
    class="c1"
    id="test-data"
  >
    <button
      aria-label="Open search"
      class="c2"
      id="test-data--search-control"
      type="button"
    >
      <svg
        aria-label="Search"
        class="c3"
        viewBox="0 0 24 24"
      >
        <path
          d="m15 15 7 7-7-7zm-5.5 2a7.5 7.5 0 1 0 0-15 7.5 7.5 0 0 0 0 15z"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`DataSearch drop 2`] = `
<button
  aria-label="Open search"
  class="StyledButton-sc-323bzc-0 emhJYD"
  data-g-tabindex="none"
  id="test-data--search-control"
  tabindex="-1"
  type="button"
>
  <svg
    aria-label="Search"
    class="StyledIcon-sc-ofa7kd-0 jQvEgu"
    viewBox="0 0 24 24"
  >
    <path
      d="m15 15 7 7-7-7zm-5.5 2a7.5 7.5 0 1 0 0-15 7.5 7.5 0 0 0 0 15z"
      fill="none"
      stroke="#000"
      stroke-width="2"
    />
  </svg>
</button>
`;

exports[`DataSearch drop 3`] = `
"@media only screen and (max-width: 768px) {
  .drdMXk {
    height: 6px;
  }
}
@media only screen and (max-width: 768px) {
  .iISQcM {
    width: 12px;
  }
}
@media only screen and (max-width: 768px) {
  .evXIdU {
    margin: 0px;
  }
}
@media only screen and (max-width: 768px) {
  .evXIdU {
    font-size: 22px;
    line-height: 28px;
    max-width: 528px;
  }
}"
`;

exports[`DataSearch renders 1`] = `
.c8 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c8 g {
  fill: inherit;
  stroke: inherit;
}

.c8 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c8 *[stroke*="#"],
.c8 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c8 *[fill-rule],
.c8 *[FILL-RULE],
.c8 *[fill*="#"],
.c8 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  height: 12px;
}

.c11 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 24px;
}

.c10 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #7D4CDB;
  color: #f8f8f8;
  border-radius: 18px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c10:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c10:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c10:focus > circle,
.c10:focus > ellipse,
.c10:focus > line,
.c10:focus > path,
.c10:focus > polygon,
.c10:focus > polyline,
.c10:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c10:focus::-moz-focus-inner {
  border: 0;
}

.c10:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c10:focus:not(:focus-visible) > circle,
.c10:focus:not(:focus-visible) > ellipse,
.c10:focus:not(:focus-visible) > line,
.c10:focus:not(:focus-visible) > path,
.c10:focus:not(:focus-visible) > polygon,
.c10:focus:not(:focus-visible) > polyline,
.c10:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c10:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c12 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  opacity: 0.3;
  cursor: default;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c12:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c12:focus > circle,
.c12:focus > ellipse,
.c12:focus > line,
.c12:focus > path,
.c12:focus > polygon,
.c12:focus > polyline,
.c12:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c12:focus::-moz-focus-inner {
  border: 0;
}

.c12:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c12:focus:not(:focus-visible) > circle,
.c12:focus:not(:focus-visible) > ellipse,
.c12:focus:not(:focus-visible) > line,
.c12:focus:not(:focus-visible) > path,
.c12:focus:not(:focus-visible) > polygon,
.c12:focus:not(:focus-visible) > polyline,
.c12:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c12:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c9 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  padding-left: 48px;
}

.c9:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c9:focus > circle,
.c9:focus > ellipse,
.c9:focus > line,
.c9:focus > path,
.c9:focus > polygon,
.c9:focus > polyline,
.c9:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c9:focus::-moz-focus-inner {
  border: 0;
}

.c9::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c9::-moz-placeholder {
  color: #AAAAAA;
}

.c9:-ms-input-placeholder {
  color: #AAAAAA;
}

.c9::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c9::-moz-focus-inner {
  border: none;
  outline: none;
}

.c9:-moz-placeholder,
.c9::-moz-placeholder {
  opacity: 1;
}

.c6 {
  position: relative;
  width: 100%;
}

.c7 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-packjustify: center;
  -webkit-justify: center;
  -ms-flex-packjustify: center;
  justify: center;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  pointer-events: none;
  left: 12px;
}

.c13 {
  opacity: 0;
}

.c2 {
  max-width: 100%;
}

.c4 {
  margin: 0px;
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

@media only screen and (max-width:768px) {
  .c5 {
    height: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    width: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    font-size: 22px;
    line-height: 28px;
    max-width: 528px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    id="data"
  >
    <form
      class="c2"
    >
      <div
        class="c1"
      >
        <header
          class="c3"
        >
          <h2
            class="c4"
          >
            Filters
          </h2>
        </header>
        <div
          class="c5"
        />
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <svg
              aria-label="Search"
              class="c8"
              viewBox="0 0 24 24"
            >
              <path
                d="m15 15 7 7-7-7zm-5.5 2a7.5 7.5 0 1 0 0-15 7.5 7.5 0 0 0 0 15z"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </div>
          <input
            aria-label="Search data"
            autocomplete="off"
            class="c9"
            id="data--search"
            name="_search"
            type="search"
            value=""
          />
        </div>
        <div
          class="c5"
        />
        <footer
          class="c3"
        >
          <button
            class="c10"
            type="submit"
          >
            Apply filters
          </button>
          <div
            class="c11"
          />
          <button
            aria-hidden="true"
            class="c12 c13"
            disabled=""
            tabindex="-1"
            type="reset"
          >
            Undo changes
          </button>
        </footer>
      </div>
    </form>
  </div>
</div>
`;
